cmd/senpai/main.go

```diff
@@ -51,16 +53,23 @@ func main() {
 	lastNetID, lastBuffer := getLastBuffer()
 	app.SwitchToBuffer(lastNetID, lastBuffer)
 
+	// Write last buffer on close
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+
+	go func() {
+		for sig := range sigCh {
+			switch sig {
+			case syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM:
```

Comment:
I'd advise against exiting on `SIGHUP`, this signal is commonly used to reload the configuration file. It'd be surprising and maybe cause conflicts if at some point senpai wants to support hot-reloading the config file.

Reply:
We could just use SIGUSR1. SIGHUP has terminal semantics, which matter in the case of a console application. Besides, we could just add /reload to reload the configuration (again because this is an interactive console application, not a background daemon).